libraries/chain/include/eos/chain/chain_controller.hpp

```diff
@@ -269,8 +270,7 @@ namespace eosio { namespace chain {
          chainbase::database& get_mutable_database() { return _db; }
 
          bool should_check_scope()const                      { return !(_skip_flags&skip_scope_check);                     }
-         bool is_producing()const                            { return _skip_flags & (received_block | pushed_transaction); }
-
+     bool is_producing()const                            { return (_skip_flags & (received_block | pushed_transaction)) && !(_skip_flags & irreversible); }
```

Comment:
We need to at least have a comment here as to what this method is about, because this indicates that we are "not in the act of producing a block", right?
Also, please fix indentation.